snpe/tutorials/marketplace_simulation.ipynb

```diff
@@ -0,0 +1,957 @@
+       "                var nbb_cells = Jupyter.notebook.get_cells();\n",
+       "                for (var i = 0; i < nbb_cells.length; ++i) {\n",
+       "                    if (nbb_cells[i].input_prompt_number == nbb_cell_id) {\n",
+       "                        if (nbb_cells[i].get_text() == nbb_unformatted_code) {\n",
+       "                             nbb_cells[i].set_text(nbb_formatted_code);\n",
+       "                        }\n",
+       "                        break;\n",
+       "                    }\n",
+       "                }\n",
+       "            }, 500);\n",
+       "            "
+      ],
+      "text/plain": [
+       "<IPython.core.display.Javascript object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "def input_generator(\n",
+    "    number_prods: int, number_users: int, artifact_path=\"../artifacts/marketplace/ \"\n",
+    "):\n",
+    "    \"\"\"\n",
+    "    This function generates and saves fictional inputs for a marketplace simulation. \n",
+    "    It requires the number of products and users to include in the fictional inputs as arguments. \n",
+    "    Output is stored by default in the path \"snpe/artifacts/marketplace/\", \n",
+    "    which is the path where the instances of the marketplace simulation-related classes are instructed \n",
+    "    to look for the inputs by default.\n",
+    "    A different path storage can be provided as the value of the optional \"artifact_path\" argument. \n",
+    "    \"\"\"\n",
+    "    assert (\n",
```

Comment:
This is because 13939 is the number of histograms included in the rating_histogram.txt file, any larger number should produce an error as there are not enough rows in the dataframe (i.e. review histograms) to assign to the generated product IDs if the number of these is larger than that.